uefi-services/src/lib.rs

```diff
@@ -104,8 +117,13 @@ fn panic_handler(info: &core::panic::PanicInfo) -> ! {
         }
     }
 
-    loop {
-        // TODO: add a timeout then shutdown.
+    if let Some(shutdown_hook) = unsafe { PANIC_SHUTDOWN_HOOK } {
+        // If the user had the time to provide a shutdown hook, run it
+        shutdown_hook();
+    } else {
+        // Otherwise, just give up and loop...
+        error!("No shutdown hook defined, will loop indefinitely...");
```

Comment:
Would be nice to also add something like "Device will have to be powered off manually" to let the user know what to do.